streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopicManager.java

```diff
@@ -83,49 +88,36 @@ public void close() {
     }
 
     /**
-     * Return the non existing topics.
-     *
-     * @param topicsPartitionsMap
-     * @param topicsMetadata
-     * @return
+     * Check the existing topics to have correct number of partitions; and return the non existing topics to be created
      */
-    private Map<InternalTopicConfig, Integer> filterExistingTopics(final Map<InternalTopicConfig, Integer> topicsPartitionsMap, Collection<MetadataResponse.TopicMetadata> topicsMetadata) {
-        Map<String, Integer> existingTopicNamesPartitions = getExistingTopicNamesPartitions(topicsMetadata);
+    private Map<InternalTopicConfig, Integer> validateTopicPartitons(final Map<InternalTopicConfig, Integer> topicsPartitionsMap,
+                                                                     final Map<String, Integer> existingTopicNamesPartitions) {
         Map<InternalTopicConfig, Integer> nonExistingTopics = new HashMap<>();
```

Comment:
I'd probably change the name of this to `topicsToBeCreated` or something similar. Also `final`